types/address/v6/compressed.pp

```diff
@@ -7,4 +7,11 @@
   /\A([[:xdigit:]]{1,4}:){5}(:|(:[[:xdigit:]]{1,4}){1,2})(\/(1([01][0-9]|[2][0-8])|[1-9][0-9]|[1-9]))?\z/,
   /\A([[:xdigit:]]{1,4}:){6}(:|(:[[:xdigit:]]{1,4}){1,1})(\/(1([01][0-9]|[2][0-8])|[1-9][0-9]|[1-9]))?\z/,
   /\A([[:xdigit:]]{1,4}:){7}:(\/(1([01][0-9]|[2][0-8])|[1-9][0-9]|[1-9]))?\z/,
+  /\Afe80:(:|(:[[:xdigit:]]{1,4}){1,6})(\/(1([01][0-9]|[2][0-8])|[1-9][0-9]|[1-9]))?%[[:lower:]]{1,8}[[:digit:]]{1,4}\z/,
```

Comment:
From RFC 4007 section 11.2,

~~~
   An implementation SHOULD support at least numerical indices that are
   non-negative decimal integers as <zone_id>.  The default zone index,
   which should typically be 0 (see Section 6), is included in the
   integers.  When <zone_id> is the default, the delimiter characters
   "%" and <zone_id> can be omitted.  Similarly, if a textual
   representation of an IPv6 address is given without a zone index, it
   should be interpreted as <address>%<default ID>, where <default ID>
   is the default zone index of the scope that <address> has.

   An implementation MAY support other kinds of non-null strings as
   <zone_id>.  However, the strings must not conflict with the delimiter
   character.  The precise format and semantics of additional strings is
   implementation dependent.
~~~

This means that, broadly, the part of the regex to match the `zone_id` should be something like

~~~
%[^%]+
~~~

or the slightly more restricted but probably sufficient

~~~
%[[:alnum:]]+
~~~

To match the minimal numerical index format as well as the variety of common network interface names, since recent Linux distros support a handful of formats (e.g., the [predictable network interface device names](https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/7/html/Networking_Guide/sec-Understanding_the_Predictable_Network_Interface_Device_Names.html)).